varname.py

```diff
@@ -182,15 +185,20 @@ def will(caller: int = 1, raise_exc: bool = True) -> Optional[str]:
     return node.attr
 
 def register(
-        cls: type = None, *,
-        caller: int = 1,
+        cls_or_func: type = None, *,
```

Comment:
I just think this whole register/inject thing is a bad idea. You've provided the essential functionality in the `varname` function - that's the part people can't do by themselves. This adds more confusion, more API to learn, and more potential for problems such as thread safety.

Reply:
Hmmm, maybe we should keep it consistent. Either just providing essential functionalities (`varname`, `nameof` and `will`) or shipping with some helper functions (`register`, `Wrapper`, etc).

Anyway, I agree that those helpers may bring potential issues.

Reply:
I am splitting the main file into different submodules, including a `helpers.py`, which will not be imported by default by `import varname`, unless someone does `import varname.helpers` and alike. I am going to move all those helper functions into that submodule and leave the choice for the people to use them or not.